packages/amplify-graphql-generation-transformer/src/grapqhl-generation-transformer.ts

```diff
@@ -0,0 +1,221 @@
+import { GenerationDirective } from '@aws-amplify/graphql-directives';
+import {
+  DirectiveWrapper,
+  TransformerPluginBase,
+  generateGetArgumentsInput,
+  TransformerResolver,
+} from '@aws-amplify/graphql-transformer-core';
+import {
+  MappingTemplateProvider,
+  TransformerContextProvider,
+  TransformerSchemaVisitStepContextProvider,
+} from '@aws-amplify/graphql-transformer-interfaces';
+import { DirectiveNode, FieldDefinitionNode, InterfaceTypeDefinitionNode, ObjectTypeDefinitionNode } from 'graphql';
+import { HttpResourceIDs, ResolverResourceIDs } from 'graphql-transformer-common';
+import { ToolConfig, createResponseTypeTool } from './utils/tools';
+import * as cdk from 'aws-cdk-lib';
+import { createInvokeBedrockResolverFunction } from './resolvers/invoke-bedrock';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { Construct } from 'constructs';
+import { validate } from './validation';
+import { toUpper } from 'graphql-transformer-common';
+
+export type GenerationDirectiveConfiguration = {
+  parent: ObjectTypeDefinitionNode;
+  directive: DirectiveNode;
```

Comment:
why do we need to store the directive definition when the actual directive arguments are already listed out separately below?

Reply:
Same applies for parent and field

Reply:
Good point, we don't! I'll remove this in a follow up. Thanks!